### DISCUSSION TITLE
Status bitrot protection

### DISCUSSION THREAD

**Discussion by omarkohl** [minio/minio#16181] [Q&A] [answered]
I am struggling a little understanding the current state of bitrot protection in MinIO.

# My understanding

* `mc admin heal` explicitly triggers a heal fixing any bitrot issues on the server, but it's deprecated because it's no longer necessary thanks to the features described below and because it puts much load on the system. Source https://github.com/minio/minio/issues/9878#issuecomment-646795214 .
* There is a background heal process that scans the entire namespace every 30 days.
  - This CAN also fix bitrot issues, but ...
  - ... it's disabled by default and needs to be enabled with `mc admin config set alias/ heal bitrotscan=on` and ...
  - ... it's NOT recommended to permanently enable it because it might make the CPU spike. I assume this is the reason it's not enabled by default. Source https://github.com/minio/minio/discussions/11664
* GET/POST requests for S3 objects automatically trigger a bitrot correction for those objects

# Questions

* Is the above correct? Am I missing something?
* Is there any other downside to enabling the background heal process with bitrotscan=on if CPU usage does not spike? i.e. is it fine to just experiment with this?
* Where can I get information about the current status of bitrot detection, were there any issues found and/or fixed etc?
  - `mc admin heal`? https://github.com/minio/minio/discussions/14257
  - I see that there is a `/background-heal/status` endpoint, but it does not seem to be referenced in the `mc` code.
  - According to https://github.com/minio/minio/discussions/11664 there are no stats, but it was being worked on and the comment is almost 2 years old. What is the current status?

**Comment by omarkohl** [comment]
Any thoughts on this?

**Reply by klauspost** [reply, marked as answer]
> Is the above correct? Am I missing something?

Seems correct to me.

> Is there any other downside to enabling the background heal process with bitrotscan=on if CPU usage does not spike? i.e. is it fine to just experiment with this?

Not other than resources usage. It will also significantly increase idle disk usage.

> current status of bitrot detection

Bitrot happens so rarely (if ever on a given system) that it doesn't really have any high priority. `mc admin trace` can also show healing events. For more details, reach out on `hello@min.io` or look through the available docs.
